src/app/core/services/notifications/fcm-rest-notification.service.ts

```diff
@@ -122,16 +123,14 @@ export class FcmRestNotificationService extends FcmNotificationService {
   }
 
   cancelAllNotifications(subject): Promise<any> {
-    return this.appServerService
-      .getHeaders()
-      .then(headers =>
-        this.http
-          .delete(
-            this.getNotificationEndpoint(subject.projectId, subject.subjectId),
-            { headers }
-          )
-          .toPromise()
+    return this.schedule.getTasks(AssessmentType.ALL).then(tasks => {
+      const fcmNotifications = this.notifications
+        .futureNotifications(tasks, DefaultNumberOfNotificationsToSchedule)
+        .filter(t => t.id)
+      return Promise.all(
+        fcmNotifications.map(n => this.cancelSingleNotification(subject, n))
       )
+    })
```

Comment:
Does this query the appserver for notifications list? OR is it just generating the future notifications based on the task schedule locally?

I think it will be better to get the list of notifications by querying the appserver, instead of using the local schedule (some may or may not be present on the appserver and some additional ones could have been added from other sources).

Reply:
It's getting the future notifications based on the local task schedule and filtering those with `IDs`. Those IDs are taken from the app server notification id upon successful scheduling. Is this fine since this will match the app server IDs? The ids are also removed when the notification is cancelled.

Reply:
I am not sure that will be enough. What happens the first time the app is opened with the new AppServer? It will not have ids from the app server, right?

Also what happens when the protocol changes, this will only delete the notifications based on new schedule but the appserver will still have notifications based on the old schedule.

And lastly, there may be some other notifications which are scheduled from other sources which the app will have no idea about (like from a UI).

I think in any conflict appserver data should be used as ground truth.

Reply:
Oh okay I see. I'll make the changes.